### PR TITLE
Created get_all_posts and get_user_post functions, and do_GET method to invoke them 

### PR DESCRIPTION
**Pull request by ericsteedly**
The purpose of this pull request is to create a post_views file and add functions for getting all posts and getting a users post from the database. do_GET method was also added to json-server.py to handle incoming requests and invoke these functions 

Both get_all_posts and get_user_posts functions return posts with nested user and category data  Ticket #52

The GET method in the JSONServer class checks the url for a user_id in the query params, if it exists it proceeds with get_user_posts. If user_id is not in the params it proceeds with get_all_posts - Ticket #60

--TESTING--

**get all posts**
Run debugger and open up postman. Set request option to GET and insert this url:

http://localhost:8088/posts

-expected outcome: Should return code 200 and all posts currently in the database 
**NOTE**
Check the SQLite database directly to confirm that all posts were indeed returned

Return format should be as follow:
```json
[
    {
        "id": 1,
        "title": "Sandwich Things",
        "publication_date": 12202020,
        "image_url": null,
        "content": "Today, I made a sandwich",
        "approved": 1,
        "user_id": 1,
        "category_id": 1,
        "category": {
            "id": 1,
            "label": "News"
        },
        "user": {
            "id": 1,
            "first_name": "Jimmy",
            "last_name": "Johns"
        }
    }
]
```

**get a user's post**
Restart debugger and insert this URL into a GET request on postman:

http://localhost:8088/posts?user=1

-expected outcome: Should return code 200 and all of the queried user's posts. 
**NOTE**
Check the SQLite database directly to confirm that the specific user's posts were indeed returned

Return format should be as follows:
```json
[
    {
        "id": 1,
        "title": "Sandwich Things",
        "publication_date": 12202020,
        "image_url": null,
        "content": "Today, I made a sandwich",
        "approved": 1,
        "user_id": 1,
        "category_id": 1,
        "category": {
            "id": 1,
            "label": "News"
        },
        "user": {
            "id": 1,
            "first_name": "Jimmy",
            "last_name": "Johns"
        }
    }
]
```

### CODE REVIEW

File: json-server.py [hunk 1]
```diff
@@ -89,6 +105,15 @@ def do_POST(self):
...
 
+
+
+
+        
+
+
+
+
+
```
Comment (Valerie-Freeman) [inline json-server.py:116]: Please remove all of this unnecessary white space, please

**Review by Valerie-Freeman — CHANGES_REQUESTED** [review]

File: json-server.py (hunk 1 above)
Comment (rachel-kendricks) [reply json-server.py:116]: dito ^

**Review by rachel-kendricks — CHANGES_REQUESTED** [review]
I was able to successfully get all users posts and get all posts from my postman request! Only thing that needs changed is deleting the blank space on the json-server.py. See Val's comment. Very nice and descriptive pull request.

File: json-server.py (hunk 1 above)
Comment (ericsteedly) [reply json-server.py:102]: Fixed and re-pushed

**Review by rachel-kendricks — APPROVED** [review]
Looks great!

**Pull request merged**